src/mca/pcompress/zlib/compress_zlib.c

```diff
@@ -60,13 +60,15 @@ static bool zlib_compress(const uint8_t *inbytes, size_t inlen, uint8_t **outbyt
     z_stream strm;
     size_t len, len2;
     uint8_t *tmp, *ptr;
+    uint32_t len3;
     int rc;
 
     /* set default output */
     *outbytes = NULL;
     *outlen = 0;
 
-    if (inlen < pmix_compress_base.compress_limit) {
+    len3 = inlen;
+    if (inlen < pmix_compress_base.compress_limit || inlen != len3) {
```

Comment:
I'm not sure I understand what these lines are supposed to accomplish. size_t is an unsigned integer - it might be 64-bit instead of 32-bit, but what difference does that make in these comparisons? We could simply test that `inlen` is less than `UINT32_MAX` - after all, if it is larger than that, then none of this works anyway...and frankly, neither will much of PMIx. It would be clearer than relying on value truncation, which can be iffy.

Reply:
I was unsure if accessing UINT32_MAX in this code base is acceptable, e.g. support for systems which do not have this constant etc. Effectively `inlen != len3` is equivalent to `inlen <= UINT32_MAX`. I have introduced the check to guarantee that inlen does not exceed this limit.

I'll adjust it to use UINT32_MAX.

Reply:
Yeah, I kinda grok'd that is what you were trying to do. I've never encountered a system that lacks that constant - if we do, we can always just define it via the configure logic.